### PR TITLE
node: processor: Fix incorrect fallthrough in `handleCleanup`

### PR DESCRIPTION
**Pull request by jynnantonix**
When processing pending observations, the `handleCleanup` function
checks if we already have a stored quorum VAA and deletes the in-memory
observation if one is found.  If a stored quorum VAA is not found, then
we're supposed to continue evaluating the other conditions and take
the appropriate action.  This was implemented using a `fallthrough`
statement.

Unfortunately, this is not how `fallthrough` works.  `fallthrough`
simply tells the compiler to execute the body of the next branch in
the switch block, *without evaluating the condition*.  It also doesn't
evaluate the conditions for any of the other branches in the switch.

In practice what this meant is that for local observations that didn't
have quorum we would always take the first branch, fall through to the
second branch, and then exit the switch.  Only once we had a quorum
(`s.submitted == true`) would we actually consider any of the other
branches in the switch.  It also meant that there was no case where we
would take the branch for re-observing messages that hadn't reached
quorum.

Fix this by moving the stored quorum VAA check into an if statement and
then falling through to the switch statement if one is not found.

This code badly needs better testing, which I will probably work on next
week.  We can either wait to merge this until I have tests ready or we can
try to merge as is if we want to get the fix out sooner.

### CODE REVIEW

File: node/pkg/processor/cleanup.go [hunk 1]
```diff
@@ -81,16 +80,17 @@ func (p *Processor) handleCleanup(ctx context.Context) {
 					p.logger.Info("Expiring late VAA", zap.String("digest", hash), zap.Duration("delta", delta))
 					aggregationStateLate.Inc()
 					delete(p.state.signatures, hash)
-					break
+					continue
```
Comment (kcsongor) [inline node/pkg/processor/cleanup.go:83]: looking at this code, `continue` _seems_ to be the right thing to do. Do you know why it did a `break` before?

**Review by kcsongor — APPROVED** [review]

File: node/pkg/processor/cleanup.go (hunk 1 above)
Comment (jynnantonix) [reply node/pkg/processor/cleanup.go:83]: It was an early break out of the switch case.

Comment (kcsongor) [reply node/pkg/processor/cleanup.go:83]: makes sense, the first resource I found on `break` only discussed loops, but but now I see that it applies to switch and select too

**Review by hendrikhofstadt — APPROVED** [review]
Great catch!

In favour of including this in the next release even without full testing. No security risk and this is clearly more correct behaviour even if the whole system still isn't 100% correct pre-tests.

**Comment by evan-gray** [comment]
Great find. Concur with getting this fix out. 🙂

**Pull request merged**